providers/implementations/kdfs/pbkdf2.c

```diff
@@ -225,6 +277,11 @@ static int kdf_pbkdf2_set_ctx_params(void *vctx, const OSSL_PARAM params[])
         if (!OSSL_PARAM_get_int(p, &pkcs5))
             return 0;
         ctx->lower_bound_checks = pkcs5 == 0;
+#ifdef FIPS_MODULE
+        ossl_FIPS_IND_set_settable(OSSL_FIPS_IND_GET(ctx),
+                                   OSSL_FIPS_IND_SETTABLE0,
+                                   ctx->lower_bound_checks);
+#endif
```

Comment:
Is this implemented correctly? 

It looks different to all other FIPS indicators.

In all other modules there is an unconditional call to

```
- OSSL_FIPS_IND_SET_CTX_PARAM(ctx, id, params, name)

This should be put at the top of the algorithms set_ctx_params().
There may be multiple of these. The name should match an
OSSL_FIPS_IND_SETTABLE_CTX_PARAM() entry.
The id should match an OSSL_FIPS_IND_ON_UNAPPROVED() entry.
```

But here the call is to ossl_FIPS_IND_set_settable; and it is conditional on PKCS5 parameter being set.

Reply:
I think it drops through to the same underlying call.  I've noticed other problems with the indicator support for PBKDF2 however.  PR coming.

Reply:
@slontis, do you remember why this indicator is different to all the rest?

I've gone through the code and it appears to be doing what's wanted and we've got sane looking evp_test indicator tests.

Reply:
ok. will test things as per documentation, to double check if fips-indicator is working as expected or not.

Reply:
@paulidale @slontis 

Hi, I struggle to see how any of the lower bounds are enforced; none of the indicators are set; and the callback is not being performed for me.

I wrote demo code and in fips mode, it appears to be successfully doing kdf with too low settings which should not have been allowed.

Please see https://github.com/openssl/openssl/issues/26983